website/docs/r/scheduled_view.html.markdown

```diff
@@ -44,6 +45,7 @@ The following arguments are supported:
 The following attributes are exported:
 
 - `id` - The internal ID of the scheduled view.
+- `index_id` - The Index ID of the scheduled view. It never updates at any point of time during resource updates, therefore make sure to ignore this via `ignore_changes = [index_id]`.
```

Comment:
This is already enforced in the API ([PUT](https://api.sumologic.com/docs/#operation/updateScheduledView) and [POST](https://api.sumologic.com/docs/#operation/createScheduledView) api docs). Here we are just communicating/informing customers as it is a non-updatable field.

Reply:
Even if the user passes `index_id`, it will simply be ignored at the API.

Reply:
>Even if the user passes index_id, it will simply be ignored at the API.

I don't think this is right behavior. We are silently dropping a field provided by the client. Instead we should return error b/c request is invalid as you are not allowed to change the index id. This is the real issue here.

It should be noted that `ignore_changes` is not strictly required if you don't change the index id which I believe most customer won't do, given we don't support it?

Reply:
> It should be noted that ignore_changes is not strictly required if you don't change the index id which I believe most customer won't do, given we don't support it?

Yes, not a strict requirement to add `ignore_changes` .. but recommended, and hence conveying that msg in the documentation here.


> I don't think this is right behavior. We are silently dropping a field provided by the client. Instead we should return error b/c request is invalid as you are not allowed to change the index id. This is the real issue here.

- This will not even reach the API layer as it'll be dropped by OpenAPI translation itself and will not translated into the request payload's Java Object. 
- Moreover, there can be some invalid fields added by the user (by mistake or intentionally), so those are also simply ignored and not translated by OpenAPI... 
- Therefore, we can't really do much with this `index_id` field being ignored.

Reply:
> We can follow-up internally on this one. I don't see a reason why OpenAPI would silently drop a field.

Sure, can I know the set of people who can help me here?

Reply:
> The language should be updated to reflect that. Currently, it seems we are asking to add ignore_changes rather than a recommendation.

But without `ignore_changes`, we are running into [this issue](https://sumologic.slack.com/archives/CAU3N0Y92/p1726892494120009?thread_ts=1726887269.983899&cid=CAU3N0Y92).. let's continue for this on the slack thread.

Reply:
> But without ignore_changes, we are running into [this issue](https://sumologic.slack.com/archives/CAU3N0Y92/p1726892494120009?thread_ts=1726887269.983899&cid=CAU3N0Y92).. let's continue for this on the slack thread.

Issue is resolved with the introduction of [computed](https://github.com/SumoLogic/terraform-provider-sumologic/pull/691/files#diff-4b1847b87c211c5fa22381b241115d5d3d299920c9b492b8ce8d669a4fcf7d41R43) prop for `index_id`.  (see [thread](https://sumologic.slack.com/archives/CAU3N0Y92/p1727231569976409?thread_ts=1726887269.983899&cid=CAU3N0Y92))

I'm updating the documentation that `ignore_changes` is just a recommendation and not a mandatory thing.